pallets/pp/src/lib.rs

```diff
@@ -267,10 +277,18 @@ impl<T: pallet_contracts::Config + pallet::Config> ChainExtension<T> for Pallet<
     where
         <E::T as SysConfig>::AccountId: UncheckedFrom<<E::T as SysConfig>::Hash> + AsRef<[u8]>,
     {
+        // Fetch AccountId of the caller from the ChainExtension's memory
+        // This way the progrmmable pool can force the caller of the ChainExtension
+        // to only spend their own funds as `ContractBalances` will be queried
+        // using `acc_id` and user cannot control the value of this variable
+        let mut env = env.buf_in_buf_out();
+        let acc_id = env.ext().address().encode();
+        let acc_id: T::AccountId = T::AccountId::decode(&mut &acc_id[..])
```

Comment:
An awful lot of acc_ids here.

Reply:
I'll come up with a better name for the first variable